### ISSUE TITLE
Allow for piggybacking of opaque tokens via the HTTP API

### ISSUE THREAD

**Issue by smn** [praekeltfoundation/vumi-go#960]
This comes from a discussion around 1 outbound message resulting in multiple delivery reports because the SMPP transport chunks a message into multiple PDUs in case it doesn't fit in a single SMS.

Some suggestions from Nic / http://rapidpro.io/:
- give every outbound message a custom callback for events (very flexible)
- aggregate all events for an outbound message and pass that back as a single "thing"
- allow for passing along of an opaque (app specific) token that gets handed back with every event for an outbound message.

**Comment by hodgestar** [comment]
Isn't the message_id in the event already an opaque token? We should
probably return the generated message_id to people who send HTTP messages
(if we don't already).

**Comment by smn** [comment]
Yeah the `user_message_id` exists but that requires the 3rd party to do the admin of that. It's doable but in some situations an opaque token that a third party owns is easier.

This was needed in Vumi Libya as well for auditing purposes where they needed to work with their own tokens.

I like the idea of allowing a custom callback URL because that allows the 3rd party developer to pass along whatever they like.
